### PR TITLE
[Platform]: aotf update require

### PR DESCRIPTION
**Pull request by chinmehta**
# [Platform]: aotf update require

## Description

- [x] Change 'Require' to 'Include'.
- [x] Add a tooltip: This results in an 'OR' filtering between the data sources.
- [x] Moving `AssociationsStateContext.jsx` functions to `aotfActions.ts` and `aotfReducer.ts` pattern
- [x] Associations `HeaderControls` folder component updated to typescript component.
- [x]  Adding a (beta) in the Search disease/target specific filter heading (facets heading)
- [x] Change a column in the facets category to Target ID (ENSG)

**Issue:** https://github.com/opentargets/issues/issues/3422
**Deploy preview:** https://deploy-preview-448--ot-platform.netlify.app/

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

- [x] Test with BE change pending

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/platform/src/components/AssociationsToolkit/components/HeaderControls/HeaderControls.tsx [hunk 1]
```diff
@@ -53,20 +88,25 @@ function HeaderControls({ cols = [] }) {
               <Typography variant="subtitle2">Weight</Typography>
             </Box>
             <Box>
-              <Typography variant="subtitle2">Require</Typography>
+              <Typography variant="subtitle2">
+                <Tooltip
+                  title="Enabling this will return either of the selected data sources."
+                  showHelpIcon
+                  style={null}
+                >
```
Comment (carcruz) [inline apps/platform/src/components/AssociationsToolkit/components/HeaderControls/HeaderControls.tsx:96]: Where this change comes from ?

File: apps/platform/src/components/AssociationsToolkit/context/aotfReducer.ts [hunk 2]
```diff
@@ -86,6 +93,54 @@ export function aotfReducer(state: State = initialState, action: Action): State
...
+      const dataSourceControls = state.dataSourceControls.map(col => {
+        if (col.aggregation === aggregation) {
+          return {
+            ...col,
+            required: !isAllActive,
+          };
+        }
+        return col;
+      });
+      console.log({
```
Comment (carcruz) [inline apps/platform/src/components/AssociationsToolkit/context/aotfReducer.ts:133]: Remove console

File: apps/platform/src/components/AssociationsToolkit/hooks/useAssociationsData.js [hunk 3]
```diff
@@ -49,7 +49,11 @@ function useAssociationsData({
           filter,
           sortBy,
           enableIndirect,
-          datasources,
+          datasources: datasources.map(el => ({
```
Comment (carcruz) [inline apps/platform/src/components/AssociationsToolkit/hooks/useAssociationsData.js:52]: This loop is unnecesary

File: apps/platform/src/components/AssociationsToolkit/static_datasets/dataSourcesAssoc.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ const dataSources = [
     weight: 1,
     isPrivate: false,
     docsLink: "https://platform-docs.opentargets.org/evidence#open-targets-genetics",
+    required: false,
```
Comment (carcruz) [inline apps/platform/src/components/AssociationsToolkit/static_datasets/dataSourcesAssoc.js:11]: I don't like this. The default required shouldn't be defined in plain meta data

**Review by carcruz — CHANGES_REQUESTED** [review]

File: apps/platform/src/components/AssociationsToolkit/components/HeaderControls/HeaderControls.tsx (hunk 1 above)
Comment (chinmehta) [reply apps/platform/src/components/AssociationsToolkit/components/HeaderControls/HeaderControls.tsx:102]: Issue: https://github.com/opentargets/issues/issues/3422

**Review by carcruz — APPROVED** [review]

**Pull request merged**